hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/KneighborAPI.java

```diff
@@ -135,7 +134,7 @@ public String post(@Context GraphManager manager,
         }
 
         long size = results.size();
-        if (request.limit != Query.NO_LIMIT && size > request.limit) {
+        if (size > request.limit) {
```

Comment:
resume `request.limit != Query.NO_LIMIT &&` to don't depend on NO_LIMIT value is LONG_MAX

Reply:
fine, the root problem is we consider not use `Long.Max` in future, otherwise it may lead a lot redundant condition & type casting